spock-core/src/main/java/org/spockframework/compiler/SpecRewriter.java

```diff
@@ -324,7 +324,7 @@ public void visitAnyBlock(Block block) {
 
     DeepBlockRewriter deep = new DeepBlockRewriter(this);
     deep.visit(block);
-    methodHasCondition |= deep.isConditionFound() || deep.isGroupConditionFound();
+    methodHasCondition |= deep.isDeepConditionFound() || deep.isGroupConditionFound();
```

Comment:
doesn't `isDeepConditionFound` suffice or is there any case where this is false bug `isGroupConditionFound` is true?

Reply:
There could actually be an empty group condition and that would then fail to compile as the generated `groupConditionFailedWithException` call uses the error collector. But as we remove the only usage of `methodHasCondition` anyway, it is pointless. :-)

Reply:
since `groupConditionFound` is only set in `conditionFound()` there shouldn't be a case where it former is true while the latter is false.

Reply:
hm, probably, but it will be gone anyway :-)